lightning/Cargo.toml

```diff
@@ -49,10 +49,14 @@ regex = { version = "1.5.6", optional = true }
 backtrace = { version = "0.3", optional = true }
 
 libm = { version = "0.2", default-features = false }
+delegate = "0.12.0"
+ext-test-macro = { path = "../ext-test-macro", optional = true }
+inventory = { version = "0.3", optional = true  }
```

Comment:
the issue is that you can't call non-const functions before `main()`, so it seems like you need to jump through the linker hacks `inventory` is doing.  and those hacks are target specific, so it wouldn't make a lot of sense to maintain them ourselves.

Reply:
a different approach would be to only inventory at the module level, requiring the `mod` to have an `xtest` macro on it.  then the inventorying can happen inside the `mod` macro invocation.

Reply:
since it's not possible to incrementally add to a global variable without `inventory`, I'm suggesting that tests are grouped into:
```rust
#[xtest]
mod tests {
...
}
```

and then the module level macro invocation will collect the tests inside the module in one swoop instead of incrementally.

This will require another level of `mod` inside the existing functional test mods.  and it will also require adding a level of indent to a couple of thousand lines or so.  Is that OK?

Reply:
it has to be a single compilation unit.  macros can't iterate on functions that are in another compilation unit AFAIK.

Reply:
actually, let me check this...

Reply:
so what's the verdict?  `inventory` or `mod`+indent everything?